libzfs.pyx

```diff
@@ -2401,19 +2401,16 @@ cdef class ZFSDataset(ZFSObject):
         for i in self.children:
             i.umount_recursive(force)
 
-    def send(self, fd, **kwargs):
+    def send(self, fd, fromname=None, toname=None, flags=set()):
```

Comment:
set() is not a singletone, should not be used as argument:

```
>>> def test(foo, flags=set()):
...     flags.add(foo)
...     print(flags)
... 
>>> test('foo')
{'foo'}
>>> test('bar')
{'bar', 'foo'}
```

Reply:
Wow, that's amazing! That makes so much sense because the set() does not get created when calling the function. Great catch!